File: alchemist/alchemist-swingui/src/main/java/it/unibo/alchemist/boundary/monitors/FXMapDisplay.java
package it.unibo.alchemist.boundary.monitors;

import it.unibo.alchemist.boundary.interfaces.FX2DOutputMonitor;
import it.unibo.alchemist.model.interfaces.Position;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class FXMapDisplay<T> extends AbstractFXDisplay<T> implements FX2DOutputMonitor<T> {
    /**
     * Default logger for this class. It hides {@link AbstractFXDisplay}'s one.
     */
    private static final Logger L = LoggerFactory.getLogger(FXMapDisplay.class);
    // TODO
    // TODO
    // TODO
    // TODO

    /**
     * Default constructor. The number of steps is set to default ({@value #DEFAULT_NUMBER_OF_STEPS}).
     */
    public FXMapDisplay() {
        super();
        // TODO
    }

    /**
     * Main constructor. It lets the developer specify the number of steps.
     *
     * @param step the number of steps
     * @see #setStep(int)
     */
    public FXMapDisplay(final int step) {
        super(step);
        // TODO
    }

    @Override
    public void zoomTo(final Position center, final double zoomLevel) {
        assert center.getDimensions() == 2;
        getWormhole().zoomOnPoint(getWormhole().getViewPoint(center), zoomLevel);
    }
}

File: alchemist/alchemist-swingui/src/main/kotlin/it/unibo/alchemist/boundary/monitors/FXMapDisplay.kt
/*
 * Copyright (C) 2010-2018, Danilo Pianini and contributors listed in the main
 * project's alchemist/build.gradle file.
 *
 * This file is part of Alchemist, and is distributed under the terms of the
 * GNU General Public License, with a linking exception, as described in the file
 * LICENSE in the Alchemist distribution's top directory.
 */

package it.unibo.alchemist.boundary.monitors

import it.unibo.alchemist.boundary.interfaces.FX2DOutputMonitor
import it.unibo.alchemist.model.interfaces.Concentration
import it.unibo.alchemist.model.interfaces.Position2D

/**
 * Simple implementation of a monitor that graphically represents a simulation on a 2D map.
 *
 * @param <T> The type which describes the [Concentration] of a molecule
</T> */
class FXMapDisplay<T>
@JvmOverloads constructor(step: Int = DEFAULT_NUMBER_OF_STEPS) : AbstractFXDisplay<T>(step), FX2DOutputMonitor<T> {
    // TODO
    // TODO
    // TODO
    // TODO

    override fun zoomTo(center: Position2D<*>, zoomLevel: Double) {
        assert(center.getDimensions() == 2)
        val wh = wormhole
        wh?.zoomOnPoint(wh.getViewPoint(center), zoomLevel)
    }

    companion object {
        /**
         * Default serial version UID.
         */
        private val serialVersionUID = 1L
    }
}
